indexer/src/repository.py

```diff
@@ -63,10 +63,22 @@ def delete_unlinked_directories(self):
         for root, dirs, files in os.walk(main_dir):
             if len(files) == 0:
                 continue
+
             # skip .DS_store files
             if len(files) == 1 and files[0].startswith("."):
                 continue
-            cur_dir = root.split(main_dir + "/")[1]
+
+            # skip the main directory itself
+            if root == main_dir:
+                continue
+
+            # extract relative path more safely
+            relative_path = os.path.relpath(root, main_dir)
+            # skip if we're still in the main directory
+            if relative_path == "." or relative_path.startswith(".."):
+                continue
+
+            cur_dir = relative_path
```

Comment:
It seems there is no need to create `relative_path` and then reassign it to `cur_dir`. Since `relative_path` isn't used anywhere else, you can assign directly to keep the code more short and readable:

```suggestion
            # extract relative path more safely
            cur_dir = os.path.relpath(root, main_dir)
            # skip if we're still in the main directory
            if cur_dir == "." or cur_dir.startswith(".."):
                continue
```